openmdao/utils/testing_utils.py

```diff
@@ -77,13 +77,14 @@ def use_tempdirs(cls):
     TestCase
         The decorated TestCase class.
     """
-    if getattr(cls, 'setUp', None):
-        setattr(cls, 'original_setUp', getattr(cls, 'setUp'))
-    setattr(cls, 'setUp', _new_setup)
-
-    if getattr(cls, 'tearDown', None):
-        setattr(cls, 'original_tearDown', getattr(cls, 'tearDown'))
-    setattr(cls, 'tearDown', _new_teardown)
+    if os.environ.get('USE_TEMPDIRS') in ['true', 'True', None]:
```

Comment:
There is a handy function that handles this sort of thing consistently across openmdao:
https://github.com/OpenMDAO/OpenMDAO/blob/master/openmdao/utils/general_utils.py#L903